WooCommerce/WooCommerceTests/Authentication/PostSiteCredentialLoginCheckerTests.swift

```diff
@@ -0,0 +1,248 @@
+                                                     roleEligibilityUseCase: roleCheckUseCase,
+                                                     stores: stores)
+        var isSuccess = false
+
+        // When
+        stores.whenReceivingAction(ofType: WordPressSiteAction.self) { action in
+            switch action {
+            case .fetchSiteInfo(_, let completion):
+                let site = Site.fake().copy(isWooCommerceActive: false)
+                completion(.success(site))
+            }
+        }
+        checker.checkEligibility(for: testURL, from: navigationController) {
+            isSuccess = true
+        }
+        waitUntil {
+            self.navigationController.presentedViewController != nil
+        }
+
+        // Then
+        XCTAssertFalse(isSuccess)
+        XCTAssertTrue(navigationController.presentedViewController is UIAlertController)
+    }
+
+    func test_error_alert_is_displayed_if_the_site_info_cannot_be_fetched() {
+        // Given
+        let applicationPassword = ApplicationPassword(wpOrgUsername: "test", password: .init("secret"))
+        let appPasswordUseCase = MockApplicationPasswordUseCase(mockGeneratedPassword: applicationPassword)
+        let roleCheckUseCase = MockRoleEligibilityUseCase()
+        let checker = PostSiteCredentialLoginChecker(applicationPasswordUseCase: appPasswordUseCase,
+                                                     roleEligibilityUseCase: roleCheckUseCase,
+                                                     stores: stores)
+        var isSuccess = false
+
+        // When
+        stores.whenReceivingAction(ofType: WordPressSiteAction.self) { action in
+            switch action {
+            case .fetchSiteInfo(_, let completion):
+                let site = Site.fake().copy(isWooCommerceActive: false)
+                completion(.failure(NetworkError.timeout))
+            }
+        }
+        checker.checkEligibility(for: testURL, from: navigationController) {
+            isSuccess = true
+        }
+        waitUntil {
+            self.navigationController.presentedViewController != nil
+        }
+
+        // Then
+        XCTAssertFalse(isSuccess)
+        XCTAssertTrue(navigationController.presentedViewController is UIAlertController)
+    }
+}
+
+private extension PostSiteCredentialLoginCheckerTests {
+    struct Constants {
+        static let eligibleRoles = ["shop_manager", "editor"]
+        static let ineligibleRoles = ["author", "editor"]
+    }
+
+    func makeUser(eligible: Bool = false) -> User {
+        User(localID: 0, siteID: 0, email: "email", username: "username", firstName: "first", lastName: "last",
+             nickname: "nick", roles: eligible ? Constants.eligibleRoles : Constants.ineligibleRoles)
+    }
+}
+
+/// MOCK: application password use case
+///
+private final class MockApplicationPasswordUseCase: ApplicationPasswordUseCase {
```

Comment:
Nit: Could we move this to a separate file and use it here and in `RequestAuthenticatorTests`?

Reply:
I'm not sure if we should import `NetworkingTests` from `WooCommerceTests`? Or should we move this code to Networking` instead? @selanthiraiyan

Reply:
Oh, I missed that these files reside in separate frameworks. Let us keep them separate. 😅